client/src/commands/run.ts

```diff
@@ -34,7 +39,23 @@ interface FoldingBlock {
   endCol: number;
 }
 
-let running = false;
+const { getState, subscribe: onRunStatusChange } = runStore;
+const { isRunning, setIsRunning } = getState();
+
+onRunStatusChange(
+  (state) => state.isRunning,
+  (newValue, oldValue) => {
+    if (newValue && !oldValue) {
+      if (isFocusLogOnExecutionStart()) {
```

Comment:
I would suggest to change this option to be "Show output channel when SAS log received", which matches current behavior on main branch.
It doesn't make sense to show output channel if no SAS log received, for example authentication failed.
It makes sense to show output channel if user opens the library pane and got session startup logs.

Reply:
any comment to this one?

Reply:
I agree with @clangsmith. I meant to say we need to wait for the first log recieved after run start, otherwise it will show even no log recieved.
Another thing is how a user choose to show the session startup log if they don't run any code but opens the library pane?

Reply:
All good feedback. https://github.com/sassoftware/vscode-sas-extension/pull/713/commits/82d208aa91034a323949f13fa131e4cfbbdc82df refactors the "on start" option to eval whether or not to show the channel only when the first chunk is written to `LogFn`